Document dictionary parameter for Search Applications (#97501)
elastic/elasticsearch: docs/reference/search-application/apis/put-search-application.asciidoc

- The template may be modified with subsequent <<put-search-application,put search application>> requests.
- If no template is specified when creating a search application, or if a template is removed from a search application, we use the <<query-string-query-ex-request,query_string>> defined in the template examples as a default.
- This template will be used by the <<search-application-search,search application search>> API to execute searches.

> - The template accepts an optional `dictionary` parameter which defines a JSON schema used for validating parameters sent to the <<search-application-search,search application search>> API

Reviewer: Is it worth adding a warning here that, while the dictionary is optional, not specifying it will not validate input into the template? Alternately if we don't want to word it that strongly, perhaps a sentence about why users may want to use the dictionary?

.Properties of `<template>`
[%collapsible%open]
=====

`script`::
(Required, object)

> The associated mustache template.

Reviewer: Should `mustache` be a link to the Mustache docs or the docs that @saarikabhasi wrote for mustache examples?

Author: We already link to the search template examples in:

https://github.com/elastic/elasticsearch/blob/4ea79d6bbd4cf4c5d01089b282f1e5848ae96dfb/docs/reference/search-application/apis/put-search-application.asciidoc?plain=1#L45

I don't think it is necessary to link to the search template examples every for every mention of templates.

`dictionary`::
(Optional, object)

> The dictionary used to validate the parameters used with the <<search-application-search, search application search>> API. The dictionary must be a valid JSON schema.

Author: thanks - I will commit this separately, since this is an asciidoc file and this change as it is will not be valid.

        "query_string": "*",
        "default_field": "*"
      }
    },
    "dictionary": {
      "properties": {
        "query_string": {
          "type": "string"
        },
        "default_field": {
          "type": "string",
          "enum": [
            "title",
            "description"
          ]
        },
        "additionalProperties": false
      },
      "required": [
        "query_string"
      ]
    }
  }
}
----
// TEST[skip:TBD]

> Given the value of the `dictionary` parameter, the <<search-application-search, search application search>> API will perform the following parameter validation:

Reviewer: ```suggestion
When the above `dictionary` parameter is specified, the <<search-application-search, search application search>> API will perform the following parameter validation:
```

* It accepts only the `query_string` and `default_field` parameters
* It verifies that `query_string` and `default_field` are both strings
* It accepts `default_field` only if it takes the values `title` or `description`

> The following example creates or updates an existing Search Application called `my-app`:

Reviewer: For this example is it worth including a sample response if input that violates the defined dictionary is sent in?
